src/ui/common/src/components/pages/workflows/components/RunWorkflowDialog.tsx

```diff
@@ -0,0 +1,276 @@
+    setParamNameToValMap({});
+  };
+
+  return (
+    <>
+      <Dialog open={open} onClose={() => setOpen(false)}>
+        <DialogTitle>Trigger a Workflow Run?</DialogTitle>
+        <DialogContent>
+          <Box sx={{ mb: 2 }}>
+            This will trigger a run of {name} immediately.
+          </Box>
+
+          {Object.keys(paramNameToDisplayProps).length > 0 && (
+            <Box>
+              <Typography sx={{ mb: 1 }} style={{ fontWeight: 'bold' }}>
+                {' '}
+                Parameters{' '}
+              </Typography>
+              <Typography variant="caption">
+                For json-serializable types like dictionaries or lists, enter
+                the string-serialized representation, without the outer quotes.
+                That is to say, the result of `json.dumps(val)`.
+              </Typography>
+            </Box>
+          )}
+          {Object.keys(paramNameToDisplayProps).map((paramName) => {
+            return (
+              <Box key={paramName}>
+                <Typography>
+                  <small>{paramName}</small>
+                </Typography>
+                <TextField
+                  fullWidth
+                  disabled={!paramNameToDisplayProps[paramName].isCustomizable}
+                  helperText={paramNameToDisplayProps[paramName].helperText}
+                  placeholder={paramNameToDisplayProps[paramName].placeholder}
+                  onChange={(e) => {
+                    paramNameToValMap[paramName] = e.target.value;
+                    setParamNameToValMap(paramNameToValMap);
+                  }}
+                  size="small"
+                />
+              </Box>
+            );
+          })}
+        </DialogContent>
+        <DialogActions>
+          <Button color="secondary" onClick={() => setOpen(false)}>
+            Cancel
+          </Button>
+          <Button color="primary" onClick={() => triggerWorkflowRun()}>
+            Run
+          </Button>
+        </DialogActions>
+      </Dialog>
+      <Snackbar
+        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+        open={showSuccessToast}
+        onClose={handleSuccessToastClose}
+        key={'workflowheader-success-snackbar'}
+        autoHideDuration={4000}
+      >
+        <Alert
+          onClose={handleSuccessToastClose}
+          severity="success"
+          sx={{ width: '100%' }}
+        >
+          {successMessage}
+        </Alert>
+      </Snackbar>
+      <Snackbar
+        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+        open={showErrorToast}
+        onClose={handleErrorToastClose}
+        key={'workflowheader-error-snackbar'}
+        autoHideDuration={4000}
+      >
+        <Alert
+          onClose={handleErrorToastClose}
+          severity="error"
+          sx={{ width: '100%' }}
+        >
+          {errorMessage}
+        </Alert>
+      </Snackbar>
+    </>
+  );
+};
+
+export { RunWorkflowDialog };
```

Comment:
I think you can directly do `export const RunWorkflowDialog: React.FC<>` or do `constant RunWorkflowDialog{...}; export default RunWorkflowDialog`. I personally prefer the latter if the file contains only one component. @agiron123 thoughts?